src/openloco/windows/KeyboardShortcuts.cpp

```diff
@@ -0,0 +1,202 @@
+        widget_end(),
+    };
+
+    namespace widx
+    {
+        enum
+        {
+            frame,
+            caption,
+            close_button,
+            panel,
+            list,
+            customize_keys_btn,
+        };
+    }
+
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+    static void on_mouse_up(window* self, widget_index i);
+    static void loc_4BE832(window* self);
+    static void tooltip(window*, widget_index);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void on_scroll_mouse_down(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index);
+
+    static void init_events()
+    {
+        _events.on_mouse_up = on_mouse_up;
+        _events.get_scroll_size = get_scroll_size;
+        _events.scroll_mouse_down = on_scroll_mouse_down;
+        _events.scroll_mouse_over = on_scroll_mouse_over;
+        _events.tooltip = tooltip;
+        _events.draw = draw;
+        _events.draw_scroll = draw_scroll;
+    }
+
+    // 0x004BE6C7
+    window* open()
+    {
+        window* window;
+
+        window = WindowManager::bringToFront(WindowType::keyboardShortcuts, 0);
+        if (window != nullptr)
+            return window;
+
+        init_events();
+
+        // 0x004BF833 (create_options_window)
+        window = WindowManager::createWindowCentred(WindowType::keyboardShortcuts, { 360, 238 }, 0, &_events);
+
+        window->widgets = _widgets;
+        window->enabled_widgets = (1 << widx::close_button) | (1 << widx::customize_keys_btn);
+        window->init_scroll_widgets();
+
+        auto skin = objectmgr::get<interface_skin_object>();
+        window->colours[0] = skin->colour_0B;
+        window->colours[1] = skin->colour_10;
+
+        window->row_count = 35;
+        window->row_hover = -1;
+
+        return window;
+    }
+
+    // 0x004BE726
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi)
+    {
+        // Draw widgets.
+        window->draw(dpi);
+    }
+    // 0x004BE72C
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+    {
+        auto colour = window->colours[1];
+        auto shade = colour::get_shade(colour, 4);
+        gfx::clear_single(*dpi, shade);
+
+        auto yPos = 0;
+        for (auto i = 0; i < window->row_count; i++)
+        {
+            string_id format = string_ids::white_stringid2;
+            if (i == window->row_hover)
+            {
+                gfx::draw_rect(dpi, 0, yPos, 800, rowHeight, 0x2000030);
+                format = string_ids::wcolour2_stringid2;
+            }
+
+            addr<0x112C826 + 2, string_id>() = ShortcutManager::getName(static_cast<Shortcut>(i));
```

Comment:
You've already defined `_commonFormatArgs`. Using it will make it easier to refactor into a format args object later.

```suggestion
            _commonFormatArgs[1] = ShortcutManager::getName(static_cast<Shortcut>(i));
```